s3_resource/src/s3_operations.cpp

```diff
@@ -2392,4 +2393,208 @@ namespace irods_s3 {
 		return SUCCESS();
 	} // s3_notify_operation
 
+    irods::error s3_read_checksum(irods::plugin_context& _ctx,
+            const std::string* _checksum_scheme,
+            std::string* _returned_checksum) {
+
+        // validate inputs are not null
+        if (!_checksum_scheme || !_returned_checksum) {
+            return ERROR(SYS_INVALID_INPUT_PARAM, "Invalid input parameter.");
+        }
+
+        irods::file_object_ptr file_obj = boost::dynamic_pointer_cast<irods::file_object>(_ctx.fco());
+
+        // create a generic not supported error for returns
+        irods::error generic_checksum_not_available_error =
+            ERROR(SYS_NOT_SUPPORTED,
+                    fmt::format("checksum scheme [{}] for [{}] is not available in S3",
+                    *_checksum_scheme,
+                    file_obj->logical_path()));
+
+
+        // if direct checksum read is not enabled, just return an error
+		if (!s3_direct_checksum_read_enabled(_ctx.prop_map())) {
+            return generic_checksum_not_available_error;
```

Comment:
Direct checksum reads not being enabled and the checksum scheme not being available in S3 are different things. I would think we would want to return a different error (or at least a different error message) for this situation?

Reply:
That's fine. I think `SYS_NOT_ALLOWED` might be an okay error, too, but I guess the important part is that an error is returned so that the FileChksum API can do something else.